pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,284 @@
+			if err := common.Rollback(cmd, k8sClient, createdRes); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Failed to rollback: %v\n", err)
+			}
+		}
+	}()
+	err = common.CreateClusterClaim(cmd, k8sClient, memberClusterNamespace, memberClusterSet, memberClusterID, &createdRes)
+	if err != nil {
+		return err
+	}
+	err = common.CreateClusterSet(cmd, k8sClient, memberClusterNamespace, memberClusterSet, joinOpts.LeaderClusterOpt.APIServer, joinOpts.LeaderClusterOpt.Secret,
+		joinOpts.MemberClusterOpt.ClusterID, joinOpts.LeaderClusterOpt.ClusterID, joinOpts.LeaderClusterOpt.Namespace, &createdRes)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for member cluster ready\n")
+	if err = waitForMemberClusterReady(cmd, k8sClient); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for member cluster ready: %v\n", err)
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Member cluster joined successfully\n")
+
+	return nil
+}
+
+func waitForMemberClusterReady(cmd *cobra.Command, k8sClient client.Client) error {
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for ClusterSet \"%s\" to be registered in leader cluster\n",
+		joinOpts.MemberClusterOpt.ClusterSetID)
+	secret := &v1.Secret{}
+	if err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: joinOpts.LeaderClusterOpt.Secret, Namespace: joinOpts.MemberClusterOpt.Namespace}, secret); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to get secret \"%s\" in Namespace %s: %v\n", joinOpts.LeaderClusterOpt.Secret, joinOpts.MemberClusterOpt.Namespace, err)
+		return err
+	}
+
+	crtData, token, err := commonarea.GetSecretCACrtAndToken(secret)
+	if err != nil {
+		return err
+	}
+	config, err := clientcmd.BuildConfigFromFlags(joinOpts.LeaderClusterOpt.APIServer, "")
+	if err != nil {
+		return err
+	}
+	config.BearerToken = string(token)
+	config.CAData = crtData
+
+	remoteClient, err := client.New(config, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	memberClusterAnnounce := "member-announce-from-" + joinOpts.MemberClusterOpt.ClusterID
+	if err := waitForMemberClusterAnnounceReady(remoteClient, memberClusterAnnounce, joinOpts.LeaderClusterOpt.Namespace); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for MemberClusterAnnounce \"%s\" in Namespace %s in leader cluster: %v\n", memberClusterAnnounce, joinOpts.LeaderClusterOpt.Namespace, err)
+		return err
+	}
+
+	if err := waitForLeaderClusterSetReady(k8sClient, joinOpts.MemberClusterOpt.ClusterSetID, joinOpts.MemberClusterOpt.Namespace, joinOpts.LeaderClusterOpt.ClusterID); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for ClusterSet \"%s\" in Namespace %s in member cluster: %v\n", joinOpts.MemberClusterOpt.ClusterSetID, joinOpts.MemberClusterOpt.Namespace, err)
+		return err
+	}
+
+	return nil
+}
+
+func waitForMemberClusterAnnounceReady(client client.Client, memberClusterAnnounce string, namespace string) error {
```

Comment:
We do not create MemberClusterAnnounce in member cluster. Why you wait for it ready in member?

Reply:
I created a remote client in a member cluster to check the MemberClusterAnnounce in the leader cluster.  In line 223. According to the discussion above, maybe I need to remove this function.

Reply:
Yes, let us remove it.